### PR TITLE
"projectile behavior"

### PR DESCRIPTION
**Pull request by Axetusk**
This has the projectile behavior you wanted me to do. Everything should be commented and the demonstration is in the sample scene. If the idea is to fire downwards then it is easy to change.

The prefab has been pushed and the requests have been fulfilled for the time being.

### CODE REVIEW

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        [SerializeField]
+        public GameObject bullet;
+
+        [SerializeField]
+        public GameObject owningObject;
+
+        void FixedUpdate()
+        {
+            GetComponent<Rigidbody2D>().velocity = this.transform.up * bulletSpeed; //Move bullet up
```
Comment (XeroKimo) [inline Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs:23]: Don't hard code the direction, actually give it a field that can be modified.

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace DBH.Runtime
+{
+    public class ProjectileBehavior : MonoBehaviour
+    {
+
+        [SerializeField]
+        private float bulletSpeed = 5f; //Bullet speed multiplier
+
+        [SerializeField]
+        public GameObject bullet;
```
Comment (XeroKimo) [inline Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs:16]: This bullet field shouldn't need to exist. The game object that this script is attached to is the bullet itself

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        void FixedUpdate()
+        {
+            GetComponent<Rigidbody2D>().velocity = this.transform.up * bulletSpeed; //Move bullet up
+        }
+
+        private void OnTriggerEnter2D(Collider2D col) //Check collision enter
+        {
+            Debug.Log("Trigger Colliders entering collision"); //Console log on collision
+            Destroy(bullet); //Destroys bullet upon contact (feel free to delete if not needed)
```
Comment (XeroKimo) [inline Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs:29]: For now just Destroy(gameObject) and not bullet due to one of the comments above

**Review by XeroKimo — CHANGES_REQUESTED** [review]
I grabbed the project, and opened up your sample scene, and the bullet prefab is not pushed, so I can't test however way you've decided to set up the scene. 

Please also make the changes as requested, and reply to anything below if you have any questions

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs (hunk 1 above)
Comment (Axetusk) [reply Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs]: Fixed it up to now have a line that assigns the direction which to fire the bullet through a field that can be modified.

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs (hunk 2 above)
Comment (Axetusk) [reply Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs]: Removed it and replaced with bulletRotation field

File: Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs (hunk 3 above)
Comment (Axetusk) [reply Assets/Delu Bullet Hell/Scripts/ProjectileBehavior.cs]: Fixed

**Review by XeroKimo — APPROVED** [review]
You're free to commit this now into the experimental branch

**Pull request merged**